FirebaseCombineSwift/Sources/Database/DatabaseReference+Combine.swift

```diff
@@ -0,0 +1,110 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#if canImport(Combine) && swift(>=5.0) && canImport(FirebaseDatabase)
+
+  import Combine
+  import FirebaseDatabase
+  import FirebaseDatabaseSwift
+
+  @available(swift 5.0)
+  @available(macOS 10.15, iOS 13, watchOS 6, tvOS 13, *)
+  public extension DatabaseReference {
+    // MARK: - Setting values on a `DatabaseReference`
+
+    /// Encodes an `Encodable` value and, on success, writes the encoded value to the database.
+    /// - Parameters:
+    ///   - value: An instance of `Encodable` to be encoded to a document.
+    ///   - encoder: An encoder instance to use to run the encoding.
+    /// - Returns: A Future that emits the result of encoding/writing the value.
+    func setValue<T: Encodable>(from encodable: T,
+                                encoder: Database.Encoder = Database.Encoder()) -> Future<
+      Void,
+      Error
+    > {
+      Future<Void, Error> { promise in
+        do {
+          try self.setValue(from: encodable, encoder: encoder, completion: { error in
+            if let error = error {
+              promise(.failure(error))
+            } else {
+              promise(.success(()))
+            }
+          })
+        } catch {
+          // NOTE: This is only correct since we know that if `try setValue` fails
+          // then the completion is not called.
+          // Would it be better/clearer to 'spell out' the setValue and basically
+          // copy what is in DatabaseReference+Decodable in order to make it clear
+          // that the `try` is connected to the encoding and not the setting of the
+          // value?
+          promise(.failure(error))
+        }
+      }
+    }
+
+    func observeSingleEvent<T: Decodable>(of eventType: DataEventType,
+                                          as type: T.Type,
+                                          decoder: Database.Decoder = Database
+                                            .Decoder()) -> Future<T, Error> {
+      Future<T, Error> { promise in
+        self.observeSingleEvent(of: eventType) { snapshot in
+          do {
+            let t = try snapshot.data(as: T.self, decoder: decoder)
+            promise(.success(t))
+          } catch {
+            promise(.failure(error))
+          }
+        }
+      }
+    }
+
+    // NOTE: Should this perhaps be named XXXPublisher(...)?
+    // If so, what should it be? 'modelPublisher', 'decodedPublisher'?
+    // Should this be built upon the snapshotPublisher, or is it an ok
+    // performance improvement to create this 'directly'?
+    func observe<T: Decodable>(_ eventType: DataEventType,
```

Comment:
TL;DR:
* keep the current naming
* and add a new method `valuePublisher` or `snapshotPublisher` that just calls `observe`

## Longer explanation:
So far, we've tried to keep the naming as close to the original API as possible, to make it easy for people to discover the Combine methods when invoking code completion.

**Hence, I think it'd be good to keep the naming as-is.**

However, it might also be useful to use XXXPublisher to keep it more in line with the work on Firestore in PR #7623.

Looking at the JavaScript implementation, they use 

* `get` for reading once
* `onValue` for reading with an observer (to read just once, they pass a parameter)

Following this, I'd suggest adding a method named `valuePublisher`, and call `observe` (i.e. this method here)


@ryanwilson and @paulb777 - I'd love to hear your thoughts on this